patching/src/main/java/org/jboss/as/patching/tool/PatchOperationTarget.java

```diff
@@ -244,58 +292,122 @@ public RemotePatchOperationTarget(PathAddress address, ModelControllerClient cli
             final ModelNode operation = createOperation(Constants.ROLLBACK, address.toModelNode(), builder);
             operation.get(Constants.PATCH_ID).set(patchId);
             operation.get(Constants.RESET_CONFIGURATION).set(resetConfiguration);
             operation.get(Constants.ROLLBACK_TO).set(rollbackTo);
-            return client.execute(operation);
+            if(streamName != null) {
+                operation.get(ModelDescriptionConstants.OP_ADDR).add(Constants.PATCH_STREAM, streamName);
+            }
+            return executeOp(operation);
         }
 
         @Override
-        protected ModelNode rollbackLast(ContentPolicyBuilderImpl builder, boolean restoreConfiguration) throws IOException {
+        protected ModelNode rollbackLast(ContentPolicyBuilderImpl builder, boolean restoreConfiguration) throws PatchingException {
+            return rollbackLast(null, builder, restoreConfiguration);
+        }
+
+        @Override
+        protected ModelNode rollbackLast(String streamName, ContentPolicyBuilderImpl builder, boolean restoreConfiguration) throws PatchingException {
             final ModelNode operation = createOperation(Constants.ROLLBACK_LAST, address.toModelNode(), builder);
             operation.get(Constants.RESET_CONFIGURATION).set(restoreConfiguration);
-            return client.execute(operation);
+            if(streamName != null) {
+                operation.get(ModelDescriptionConstants.OP_ADDR).add(Constants.PATCH_STREAM, streamName);
+            }
+            return executeOp(operation);
+        }
+
+        @Override
+        protected ModelNode info(String patchId, boolean verbose) throws PatchingException {
+            return info(null, patchId, verbose);
         }
 
         @Override
-        protected ModelNode info(String patchId, boolean verbose) throws IOException {
+        protected ModelNode info(String streamName, String patchId, boolean verbose) throws PatchingException {
             final ModelNode operation = new ModelNode();
-            operation.get(ModelDescriptionConstants.OP).set("patch-info");
+            operation.get(ModelDescriptionConstants.OP).set(Constants.PATCH_INFO);
             operation.get(ModelDescriptionConstants.OP_ADDR).set(address.toModelNode());
             operation.get(Constants.PATCH_ID).set(patchId);
+            if(streamName != null) {
+                operation.get(ModelDescriptionConstants.OP_ADDR).add(Constants.PATCH_STREAM, streamName);
+            }
             if(verbose) {
                 operation.get(Constants.VERBOSE).set(true);
             }
-            return client.execute(operation);
+            return executeOp(operation);
+        }
+
+        protected ModelNode executeOp(ModelNode operation) throws PatchingException {
+            try {
+                return client.execute(operation);
+            } catch (IOException e) {
+                throw new PatchingException("Failed to execute operation " + operation, e);
```

Comment:
Do we care about i18n in the tool? I suppose it's governed by the same rules as the CLI.